crates/net/downloaders/src/bodies/task.rs

```diff
@@ -120,9 +120,9 @@ impl<T: BodyDownloader> Future for SpawnedDownloader<T> {
                     if let Err(err) = this.downloader.set_download_range(range) {
                         tracing::error!(target: "downloaders::bodies", ?err, "Failed to set bodies download range");
 
-                        let forward_error_result = ready!(this.bodies_tx.poll_reserve(cx))
-                            .and_then(|_| this.bodies_tx.send_item(Err(err)));
-                        if forward_error_result.is_err() {
+                        // Clone the sender ensure its availability. See [PollSender::clone].
+                        let mut bodies_tx = this.bodies_tx.clone();
+                        if bodies_tx.send_item(Err(err)).is_err() {
```

Comment:
this will panic, still needs a poll_reserve call

Reply:
even though we know for sure it's either Idle or Closed? interesting...